src/System.Slices/src/System/Span.cs

```diff
@@ -333,63 +323,15 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
-        /// Returns an enumerator over the span's entire contents.
-        /// </summary>
-        public Enumerator GetEnumerator()
-        {
-            return new Enumerator(this);
-        }
-
-        IEnumerator IEnumerable.GetEnumerator()
-        {
-            return GetEnumerator();
-        }
-
-        IEnumerator<T> IEnumerable<T>.GetEnumerator()
-        {
-            return GetEnumerator();
-        }
-
-        /// <summary>
-        /// A struct-based enumerator, to make fast enumerations possible.
-        /// This isn't designed for direct use, instead see GetEnumerator.
+        /// Returns item from given index without the boundaries check
+        /// use only in places where moving outside the boundaries is impossible
+        /// gain: performance: no boundaries check (single operation) 
         /// </summary>
-        public struct Enumerator : IEnumerator<T>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private T GetItemWithoutBoundariesCheck(int index)
```

Comment:
Should this be public + unsafe? We don't want people to do any hacks to achieve the same thing